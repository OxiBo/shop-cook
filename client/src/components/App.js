import React, { Component } from "react";
import styled, { ThemeProvider, createGlobalStyle } from "styled-components";
import theme from "./styles/theme";

const GlobalStyle = createGlobalStyle`
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  /* font-size: 15px; */

}
*, *:before, *:after{
  box-sizing: inherit;
}
html{
  font-size: 60%;
  @media only screen and (max-width: 68.75em) {
    html {
      font-size: 50%; } }
}
body {
  box-sizing: border-box;
  background-image: linear-gradient(to right bottom, ${(props) =>
    props.theme.mainGradientFrom}, ${(props) => props.theme.mainGradientTo});
    background-size: cover;
    background-repeat: no-repeat;
  /* font-size: 12px; */
  min-height: calc(100vh - 2 * 4vw);
}`;

const Container = styled.div`
  margin: 0 auto;
  max-width: 120rem;
`;

class App extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <Container>
          <GlobalStyle />
          {this.props.children}
        </Container>
      </ThemeProvider>
    );
  }
}
export default App;
