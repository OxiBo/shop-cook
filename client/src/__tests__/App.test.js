import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";

import App from "../components/App";

describe("<App />", () => {
  it("Renders the app properly", async () => {
    const wrapper = mount(<App />);
    await wait();
    wrapper.update()
    // console.log(wrapper.debug());
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
