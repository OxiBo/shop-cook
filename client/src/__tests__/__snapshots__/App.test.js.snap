// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders the app properly 1`] = `
<App>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "mistyrose",
        "backgroundMain": "green",
        "fontSize": "1.5rem",
        "headerColor": "#F9F5F3",
        "mainButtonGradientFrom": "#fbdb89",
        "mainButtonGradientTo": "#f48982",
        "mainContainerColor": "beige",
      }
    }
  >
    <div>
      <Memo(GlobalStyleComponent) />
    </div>
  </ThemeProvider>
</App>
`;
